java-manta-client/src/main/java/com/joyent/manta/client/crypto/EncryptingEntityHelper.java

```diff
@@ -83,12 +88,15 @@ public static OutputStream makeCipherOutputForStream(
      * wrapped requires changes to EncryptionStateRecorder!
      *
      * @param httpOut output stream for writing to the HTTP network socket
-     * @param encryptionContext current encryption running state
+     * @param cipherDetails current encryption running state's cipher details
+     * @param cipher current encryption running state's cipher
      * @param hmac current HMAC object with the current checksum state
      * @return a new stream configured based on the parameters
      */
     public static OutputStream makeCipherOutputForStream(
-            final OutputStream httpOut, final EncryptionContext encryptionContext,
+            final OutputStream httpOut,
+            final SupportedCipherDetails cipherDetails,
+            final Cipher cipher,
```

Comment:
Not a blocker for this PR but do we want to validate that the hmac is not null when the `CipherDetails` indicates the cipher is not AEAD?